test-suite/README.new.md

```diff
@@ -0,0 +1,324 @@
+161006 |         ../modules/Nat.v)
+161007 |   (action
+161008 |    (run %{bin:coqc} -bt -R .. Mods
+161009 |     Nat_debug_f63227d67eca844cbf7b40e6a4d45f04.v)))
+cannot guess a path for Coq libraries; please use -coqlib option or ensure you have installed the package containing Coq's stdlib (coq-stdlib in OPAM) If you intend to use Coq without a standard library, the -boot -noinit options must be used.
+```
+
+If a failure is due to a missing dependency for a test, make sure you ran `make
+dunestrap` so that the test-suite rules are fresh. If it still doesn't work,
+open an issue or complain on Zulip.
+
+## Adding a Test
+
+For simple tests, placing the new `.v` file in the correct directory will be
+enough. If the test expects an output, a file `test.out` should provided for the
+file `test.v`. The `.out` file can be omitted for the time being, and the
+correct output generated using `dune promote`. See [Promoting test
+output](#promoting-test-output).
+
+### Promoting Test Output
+
+If the ouput of a test does not match the expected output, Dune will report the
+diff. You can use `make promote` in order to promote all the actual output files
+form the `_build` directory to be the expected output.
+
+ To have finer control, use `dune promote` on a specific target such as
+ ```sh
+ dune promote test-suite/bugs/mybug.out
+ ```
+
+### Output Test Example
+
+Adding `_test.v` to `output/` for instance will yield the following when running
+`make test`:
+```sh
+Error: No rule found for test-suite/output/_test.out
+-> required by alias test-suite/output/runtest in
+   test-suite/test_suite_rules.sexp:56694
+make: *** [Makefile:122: test] Error 1
+```
+To fix this, we `touch` the file:
+```sh
+touch test-suite/output/_test.out
+```
+and rerun `make test`. Now Dune will complain that the output is different:
+```diff
+File "test-suite/output/_test.out", line 1, characters 0-0:
+diff --git a/_build/default/test-suite/output/_test.out b/_build/default/test-suite/output/_test.v.log
+index e69de29bb2..a7de56ea32 100644
+--- a/_build/default/test-suite/output/_test.out
++++ b/_build/default/test-suite/output/_test.v.log
+@@ -0,0 +1,2 @@
++Type : Type
++     : Type
+make: *** [Makefile:122: test] Error 1
+```
+Running `make promote` will now update the `.out` file with the new output:
+```sh
+Promoting _build/default/test-suite/output/_test.v.log to
+  test-suite/output/_test.out.
+```
+Finally when running `make test`, Dune will report no problems.
+
+## Running Single Tests
+
+All tests output targets such as .vo files or .log files. You can tell Dune to
+build these targets directly. For example:
+
+```sh
+dune build test-suite/bugs/bug_1446.vo
+```
+
+However this is not ideal, since it requires you to know exactly where the test
+is located and at least one of the targets it is expected to produce.
+
+To remedy this, the test-suite will provide a *Dune alias* for each of the `.v`
+file tests. Simply write:
+
+```sh
+dune build @bug_1446
```

Comment:
When tests in separate directories have the same name they end up sharing the alias.
Is it possible to have the alias be `@bugs/bug_1446` or some such thing instead?

Reply:
That alias will already work.